rpcs3/Emu/RSX/GL/GLGSRender.cpp

```diff
@@ -877,6 +880,15 @@ void GLGSRender::on_init_thread()
 
 		m_frame->enable_wm_event_queue();
 		m_shaders_cache->load(&helper);
+
+		if (g_cfg.video.perf_overlay.perf_overlay_enabled)
+		{
+			m_perf_overlay = std::make_unique<rsx::overlays::perf_metrics_overlay>(false);
```

Comment:
With the refactor now merged, you no longer need to worry about managing objects so m_perf_overlay variable should not be needed anymore. Also this code now belongs in rsx::thread::on_task in the block with the make_always<rsx::overlays::display_manager>. Its drawing is also handled automatically so no need for any code in the renderers at all, just set and forget.

Reply:
Alternatively, move this to rsx_utils, where I eventually plan on moving all remaining traces of overlays to remove them from the core files completely.